### PR TITLE
Update Azure Database Migration Service to add Managed Instance as target

### PR DESCRIPTION
**Pull request by hitenjava**
Swagger spec - https://github.com/Azure/azure-rest-api-specs/tree/master/specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-03-15-preview

PR link - https://github.com/Azure/azure-rest-api-specs/pull/2531

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please run `msbuild build.proj /t:build /p:Scope=Sdks/DataMigration` which should generate a `.props` file. Commit this file

**Comment by hitenjava** [comment]
@dsgouda , I tried running:
 msbuild build.proj /t:build /p:Scope=Sdks/DataMigration.

The build succeeded but it did not generate .props file.

File: src/SDKs/DataMigration/Management.DataMigration/Generated/Models/ValidationStatus.cs [hunk 1]
```diff
@@ -10,87 +10,19 @@
 
 namespace Microsoft.Azure.Management.DataMigration.Models
 {
-    using Newtonsoft.Json;
```
[the remark this answers is not in the document]
Comment (hitenjava) [reply src/SDKs/DataMigration/Management.DataMigration/Generated/Models/ValidationStatus.cs]: Fixed

File: src/SDKs/_metadata/datamigration_resource-manager.txt [hunk 2]
```diff
@@ -1,11 +1,10 @@
-2017-11-15 22:31:12 UTC
+2018-02-28 21:09:33 UTC
 
 1) azure-rest-api-specs repository information
 GitHub user: Azure
-Branch:      current
-Commit:      5e06f63cdec58505463668c21552e54d9d255870
+Branch:      master
```
[the remark this answers is not in the document]
Comment (hitenjava) [reply src/SDKs/_metadata/datamigration_resource-manager.txt]: Fixed.

**Comment by dsgouda** [comment]
Hmm it should, can you try this instead
Please make sure you have run
```
msbuild build.proj
```
this installs all the tools required for building this repo
```
 msbuild build.proj /t:build /p:Scope=SDKs/DataMigration
```

**Comment by hitenjava** [comment]
@dsgouda That worked. Added the props file.

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj [hunk 3]
```diff
@@ -5,8 +5,8 @@
 
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.DataMigration</PackageId>
-		<Description>Provides developers with libraries for the Data Migration Service (DMS) under Azure Resource manager to deploy an instance of DMS. Create, Delete, Create project, Start Migration, and more.</Description>
-		<Version>0.1.2-preview</Version>
+		<Description>Provides developers with libraries for the Database Migration Service (DMS) under Azure Resource manager to deploy an instance of DMS. Create, Delete, Create project, Start Migration, and more.</Description>
+		<Version>0.2.1-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: nit: versioning starts from `0`

**Review by dsgouda — APPROVED** [review]
LGTM apart from the minor comment.

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj (hunk 3 above)
Comment (hitenjava) [reply src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: Fixed

**Comment by hitenjava** [comment]
Please let me know if anything else is needed from my side to get this merged and published.
Scope - SDKs\DataMigration
Package Name - Microsoft.Azure.Management.DataMigration

**Comment by dsgouda** [comment]
@shahabhijeet PTAL and merge

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**